datastore/db/bytebase/FertiScan/delete_inspection_function.sql

```diff
@@ -0,0 +1,125 @@
+-- Function to delete organization_information
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".delete_organization_information(
+    p_organization_information_id uuid
+)
+RETURNS void
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    DELETE FROM "fertiscan_0.0.13".organization_information
+    WHERE id = p_organization_information_id;
+END;
+$$;
+
+-- Trigger function to handle after organization_information delete for location deletion
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig()
+RETURNS TRIGGER
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    IF OLD.location_id IS NOT NULL THEN
+        DELETE FROM "fertiscan_0.0.13".location
+        WHERE id = OLD.location_id;
+    END IF;
+
+    RETURN NULL;
+END;
+$$;
+
+-- Trigger definition on organization_information table for location deletion
+DROP TRIGGER IF EXISTS after_organization_information_delete_location ON "fertiscan_0.0.13".organization_information;
+CREATE TRIGGER after_organization_information_delete_location
+AFTER DELETE ON "fertiscan_0.0.13".organization_information
+FOR EACH ROW
+EXECUTE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig();
```

Comment:
Is there instances where 2 organization might ahve the same location? If so, it might cause confloct for the organization information with said location.

Reply:
Yes. This is now handled by catching possible foreign key exception and raising a notice.

Reply:
I dont see us handling the following use case: 

organization_A uses location_A
organization_B uses location_A

From my understanding, if you delete organization_A, you also delete location_A. 
However you shouldn't since organization_B also uses location_A.

If this case is handled, please point me towards where, because I can't seem to find it 🫤

Reply:
In the schema, note the fk constraint:
```sql
    CREATE TABLE "fertiscan_0.0.13"."organization_information" (
        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
        "name" text,
        "website" text,
        "phone_number" text,
        "location_id" uuid REFERENCES "fertiscan_0.0.13".location(id),  <--- here
        "edited" boolean DEFAULT false
    );
```


In the `after_org_info_delete_location_trig` trigger function, note that I raise a simple notice in case there is a fk error:
```sql
CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig()
RETURNS TRIGGER
LANGUAGE plpgsql
AS $$
BEGIN
    IF OLD.location_id IS NOT NULL THEN
        BEGIN
            DELETE FROM "fertiscan_0.0.13".location
            WHERE id = OLD.location_id;
        EXCEPTION WHEN foreign_key_violation THEN <--- from here
            RAISE NOTICE 'Location % is still referenced by another record and cannot be deleted.', OLD.location_id;
        END;
    END IF;

    RETURN NULL;
END;
$$;

```

So in your use case, if organization_A is deleted, it should not delete location because it is organization_A that has location's id. But even in the advent the location's deletion is attempted, if it is related to any organization_info, a fk exception will be raised. And in the case of `after_org_info_delete_location_trig`, the exception will just be silenced.